doc/adr/0004-nats-headers.md

```diff
@@ -92,3 +129,82 @@ Implemented and merged to master.
 ## Consequences
 
 Use of headers is possible.
+
+## Compatibility Across NATS Clients
+
+The following is a list of features to insure compatibility across NATS clients
+that support headers. Because the feature in Go client and nats-server leverage
+the Go implementation as described above, the API used will determine how header
+names are serialized.
+
+The Go implementation of `http.Header` is a `map[string][]string` If the `Set`,
+`Get`, `Values`, `Del` functions are used, the specified key arguments will be
+converted into their canonical format. This means that the serialized Header
```

Comment:
No, we no longer use that. The nats message has now its own Header type, so Set()/Add()/etc.. work directly with user key and don't do any modification.

Reply:
removed all references to the go library